pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/TransformFunctionFactory.java

```diff
@@ -281,6 +282,13 @@ public static TransformFunction get(ExpressionContext expression, Map<String, Co
         List<ExpressionContext> arguments = function.getArguments();
         int numArguments = arguments.size();
 
+        // Check if the function is ArrayLiteraltransform function
+        if (functionName.equalsIgnoreCase(ArrayLiteralTransformFunction.FUNCTION_NAME)) {
+          return queryContext.getOrComputeSharedValue(ArrayLiteralTransformFunction.class,
+              expression.getFunction().getArguments(),
```

Comment:
Check if the arguments are all list of literal contexts either here or inside the constructor

Reply:
done.